server/models/idea.js

```diff
@@ -0,0 +1,45 @@
+const MongoClient = require('mongodb').MongoClient;
+const url = 'mongodb://localhost:27017/rankerdb';
+
+exports.all = function(callback) {
+	MongoClient.connect(url, function(err, db) {
+	  if(err)throw(err);
+
+  	db.collection('ideas').find().toArray(function(err, result) {
+  		if (err) throw err;
+  		callback(result);
+  	});
+	});
+};
+
+exports.get = function(name, callback) {
+	MongoClient.connect(url, function(err, db) {
+	  if(err)throw(err);
+
+	  db.collection('ideas').find({name: name}).toArray(function(err, result) {
+	    if (err) throw err;
+	    callback(result);
+	  });
+	});
+};
+
+exports.new = function (name, desc, rating) {
+	MongoClient.connect(url, function(err, db) {
+	  if(err)throw(err);
+	  db.collection('ideas').insertOne({name: name, desc: desc, rating: rating});
```

Comment:
It would probably be a good idea to return the _id field when creating a new idea. That way the webapp can immediately start interacting with the newly created document.